src/main/java/com/aws/greengrass/secretmanager/SecretManager.java

```diff
@@ -100,16 +113,24 @@
         this.secretDao = dao;
         this.secretClient = secretClient;
         this.crypter = crypter;
+        this.initialized.set(null);
     }
 
     /**
      * Syncs secret manager by downloading secrets from cloud and then stores it locally.
      * This is used when configuration changes and secrets have to be re downloaded.
      * @param configuredSecrets List of secrets that are to be downloaded
      * @throws SecretManagerException when there are issues reading/writing to disk
+     * @throws InterruptedException if thread is interrupted while running
      */
-    public void syncFromCloud(List<SecretConfiguration> configuredSecrets) throws SecretManagerException {
+    public void syncFromCloud(List<SecretConfiguration> configuredSecrets)
+            throws SecretManagerException, InterruptedException {
         logger.atDebug("sync-secret-from-cloud").log();
+        try {
+            waitForInitialization();
```

Comment:
Do you still need to wait here?

Reply:
Yes we do. This can be called from another thread which would cause it to potentially not be initialized